src/main.c

```diff
@@ -1,20 +1,42 @@
 #include <utils/logger.h>
 #include <queue.h>
+#include <stdatomic.h>
+#include <stdbool.h>
 #include <stdio.h>
 #include <unistd.h>
 #include <signal.h>
 #include <producer.h>
 #include <utils/helper.h>
 
-int run   = 1;
-int ready = 0;
+
+/* Schaufel keeps track of consume and produce states.
+ *
+ * consume_state is statically initialised to true (for convenience).
+ * produce_state is initialised on runtime when at least one producer is up.
+ *
+ * The macros must be used to get & set the values of respective states.
+ *
+ * For p = {consume,produce}_process,
+ * $p_init() sets the state to true
+ * $p_done() sets the state to false
+ * $p_stat() queries the value of the state
+ */
+static volatile atomic_bool consume_state = ATOMIC_VAR_INIT(true);
+static volatile atomic_bool produce_state /* = ATOMIC_VAR_INIT(false) */;
```

Comment:
why the comment?

Reply:
For clarity. Since `atomic_flag` is not auto-initialised to zero unlike `atomic_bool` I thought it would make sense to comment here to prevent any possible confusion.

That said I don't think it makes a big difference, so I'll drop that.